### ISSUE TITLE
Xiaomi Mijia LSC-M01 has no image

### ISSUE THREAD

**Issue by zzaal** [al-one/hass-xiaomi-miot#568]
Hello!
After updating from 0.5.8 to 0.6.1 (0.6.2), Xiaomi Mijia LSC-M01 has no image (entity_id: camera.loock_v02_1ae3_video_doorbell).
Rollback to 0.5.8 solves the problem.
Please fix it in the next versions.

**Comment by al-one** [comment]
Show me the [state attributes](https://my.home-assistant.io/redirect/developer_states) of this entity.

**Comment by zzaal** [comment]
access_token: *******************************************
brand: loock
frontend_stream_type: hls
model: loock.cateye.v02
lan_ip: xxx.xxx.xxx.xxx
mac_address: xx:xx:xx:xx:xx:xx
entity_class: MiotCameraEntity
home_room: Home
miot_type: urn:miot-spec-v2:device:video-doorbell:0000A03A:loock-v02:1
video_doorbell.night_shot: 0
video_doorbell.wdr_mode: true
image_distortion_correction-4-3: true
motion_detection: true
detection_sensitivity-5-2: 2
battery.battery_level: 52
state_updater: cloud
motion_video_time: 2022-05-04 19:15:51
motion_video_type: Pass
stream_address: https://business.smartcamera.api.io.mi.com/common/app/m3u8?data=**********************************************************************************************************************************************************************************************************************************************************************
image_address: https://processor.smartcamera.api.io.mi.com/miot/camera/app/v1/img?data=**********************************************************************************************************************************************************************************************************************************************************************
sub_entities: motion_detection-5.motion_detection-1, battery-3.battery_level-1
motion_video_latest: 
startDuration: 0
cut: false
offset: 0
isCloud: false
isRead: false
videoStoreId: >-
  ****************************************************************************************************************************************************************************************************************************************************************************
eventType: Pass
expireTime: 1651940221677
isAlarm: true
imgStoreId: >-
  ****************************************************************************************************************************************************************************************************************************************************************************
fileId: '73986654430831872'
extraInfo: "{\n\t\"is_over\":\t1,\n\t\"video_time\":\t1651680951,\n\t\"push_type\":\t0\n}"

entity_picture: /api/camera_proxy/camera.loock_v02_1ae3_video_doorbell?token=***********************************************************************
friendly_name: Eye_Cat Video Doorbell
supported_features: 2

**Comment by kodbuzz** [comment]
Having the same issue with:

 - madv.cateye.miowl3
 - chuangmi.camera.055a02

I understand that many models won't work properly due to FFMPEG and probably non-standard stream issue, but right after updating from 0.5.8 to 0.6.x, the only sweet part for these unblessed cameras fell into failure, as still image is broken.

HA Core: 2022.5.0
Supervisor: 2022.05.0

Log - madv.cateye.miowl3:

> Update for camera.madv_miowl3_xxxx_video_doorbell fails
> Traceback (most recent call last):
>   File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 386, in _make_request
>     self._validate_conn(conn)
>   File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 1040, in _validate_conn
>     conn.connect()
>   File "/usr/local/lib/python3.9/site-packages/urllib3/connection.py", line 414, in connect
>     self.sock = ssl_wrap_socket(
>   File "/usr/local/lib/python3.9/site-packages/urllib3/util/ssl_.py", line 449, in ssl_wrap_socket
>     ssl_sock = _ssl_wrap_socket_impl(
>   File "/usr/local/lib/python3.9/site-packages/urllib3/util/ssl_.py", line 493, in _ssl_wrap_socket_impl
>     return ssl_context.wrap_socket(sock, server_hostname=server_hostname)
>   File "/usr/local/lib/python3.9/ssl.py", line 500, in wrap_socket
>     return self.sslsocket_class._create(
>   File "/usr/local/lib/python3.9/ssl.py", line 1040, in _create
>     self.do_handshake()
>   File "/usr/local/lib/python3.9/ssl.py", line 1309, in do_handshake
>     self._sslobj.do_handshake()
> socket.timeout: _ssl.c:1112: The handshake operation timed out
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/usr/local/lib/python3.9/site-packages/requests/adapters.py", line 440, in send
>     resp = conn.urlopen(
>   File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 785, in urlopen
>     retries = retries.increment(
>   File "/usr/local/lib/python3.9/site-packages/urllib3/util/retry.py", line 550, in increment
>     raise six.reraise(type(error), error, _stacktrace)
>   File "/usr/local/lib/python3.9/site-packages/urllib3/packages/six.py", line 770, in reraise
>     raise value
>   File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 703, in urlopen
>     httplib_response = self._make_request(
>   File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 389, in _make_request
>     self._raise_timeout(err=e, url=url, timeout_value=conn.timeout)
>   File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 340, in _raise_timeout
>     raise ReadTimeoutError(
> urllib3.exceptions.ReadTimeoutError: HTTPSConnectionPool(host='business.smartcamera.api.io.mi.com', port=443): Read timed out. (read timeout=10)
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 515, in async_update_ha_state
>     await self.async_device_update()
>   File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 751, in async_device_update
>     await task
>   File "/config/custom_components/xiaomi_miot/camera.py", line 291, in async_update
>     rdt = await mic.async_request_api(api, rqd, method='GET', crypt=True) or {}
>   File "/config/custom_components/xiaomi_miot/core/xiaomi_cloud.py", line 158, in async_request_api
>     return await self.hass.async_add_executor_job(
>   File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 58, in run
>     result = self.fn(*self.args, **self.kwargs)
>   File "/config/custom_components/xiaomi_miot/core/xiaomi_cloud.py", line 170, in request_miot_api
>     rsp = self.request_rc4_api(api, params, method, **kwargs)
>   File "/config/custom_components/xiaomi_miot/core/xiaomi_cloud.py", line 529, in request_rc4_api
>     response = self.session.get(url, params=params, timeout=timeout)
>   File "/usr/local/lib/python3.9/site-packages/requests/sessions.py", line 542, in get
>     return self.request('GET', url, **kwargs)
>   File "/usr/local/lib/python3.9/site-packages/requests/sessions.py", line 529, in request
>     resp = self.send(prep, **send_kwargs)
>   File "/usr/local/lib/python3.9/site-packages/requests/sessions.py", line 645, in send
>     r = adapter.send(request, **kwargs)
>   File "/usr/local/lib/python3.9/site-packages/requests/adapters.py", line 532, in send
>     raise ReadTimeout(e, request=request)
> requests.exceptions.ReadTimeout: HTTPSConnectionPool(host='business.smartcamera.api.io.mi.com', port=443): Read timed out. (read timeout=10)

Log - chuangmi.camera.055a02:

> Update for camera.chuangmi_055a02_xxxx_camera_control fails
> Traceback (most recent call last):
>   File "/usr/local/lib/python3.9/site-packages/urllib3/connection.py", line 174, in _new_conn
>     conn = connection.create_connection(
>   File "/usr/local/lib/python3.9/site-packages/urllib3/util/connection.py", line 72, in create_connection
>     for res in socket.getaddrinfo(host, port, family, socket.SOCK_STREAM):
>   File "/usr/local/lib/python3.9/socket.py", line 954, in getaddrinfo
>     for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
> socket.gaierror: [Errno -3] Try again
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 703, in urlopen
>     httplib_response = self._make_request(
>   File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 386, in _make_request
>     self._validate_conn(conn)
>   File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 1040, in _validate_conn
>     conn.connect()
>   File "/usr/local/lib/python3.9/site-packages/urllib3/connection.py", line 358, in connect
>     self.sock = conn = self._new_conn()
>   File "/usr/local/lib/python3.9/site-packages/urllib3/connection.py", line 186, in _new_conn
>     raise NewConnectionError(
> urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPSConnection object at 0x5b8a3670>: Failed to establish a new connection: [Errno -3] Try again
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/usr/local/lib/python3.9/site-packages/requests/adapters.py", line 440, in send
>     resp = conn.urlopen(
>   File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 785, in urlopen
>     retries = retries.increment(
>   File "/usr/local/lib/python3.9/site-packages/urllib3/util/retry.py", line 592, in increment
>     raise MaxRetryError(_pool, url, error or ResponseError(cause))
> urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='business.smartcamera.api.io.mi.com', port=443): Max retries exceeded with url: /common/app/get/eventlist?data=#redacted# (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x5b8a3670>: Failed to establish a new connection: [Errno -3] Try again'))
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 515, in async_update_ha_state
>     await self.async_device_update()
>   File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 743, in async_device_update
>     raise exc
>   File "/config/custom_components/xiaomi_miot/camera.py", line 291, in async_update
>     rdt = await mic.async_request_api(api, rqd, method='GET', crypt=True) or {}
>   File "/config/custom_components/xiaomi_miot/core/xiaomi_cloud.py", line 158, in async_request_api
>     return await self.hass.async_add_executor_job(
>   File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 58, in run
>     result = self.fn(*self.args, **self.kwargs)
>   File "/config/custom_components/xiaomi_miot/core/xiaomi_cloud.py", line 170, in request_miot_api
>     rsp = self.request_rc4_api(api, params, method, **kwargs)
>   File "/config/custom_components/xiaomi_miot/core/xiaomi_cloud.py", line 529, in request_rc4_api
>     response = self.session.get(url, params=params, timeout=timeout)
>   File "/usr/local/lib/python3.9/site-packages/requests/sessions.py", line 542, in get
>     return self.request('GET', url, **kwargs)
>   File "/usr/local/lib/python3.9/site-packages/requests/sessions.py", line 529, in request
>     resp = self.send(prep, **send_kwargs)
>   File "/usr/local/lib/python3.9/site-packages/requests/sessions.py", line 645, in send
>     r = adapter.send(request, **kwargs)
>   File "/usr/local/lib/python3.9/site-packages/requests/adapters.py", line 519, in send
>     raise ConnectionError(e, request=request)
> requests.exceptions.ConnectionError: HTTPSConnectionPool(host='business.smartcamera.api.io.mi.com', port=443): Max retries exceeded with url: /common/app/get/eventlist?data=#redacted# (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x5b8a3670>: Failed to establish a new connection: [Errno -3] Try again'))

**Comment by al-one** [comment]
> image_address: https://processor.smartcamera.api.io.mi.com/miot/camera/app/v1/img?data=***
> entity_picture: /api/camera_proxy/camera.loock_v02_1ae3_video_doorbell?token=***

You can add a picture card to display the motion picture of the video doorbell.

**Comment by kodbuzz** [comment]
> 

Thanks pal, just I guess I have done it already, like this:

> show_state: true
> show_name: true
> camera_view: auto
> type: picture-entity
> entity: camera.madv_miowl3_xxxx_video_doorbell
> entities:
>   - sensor.madv_miowl3_xxxx_battery_level
> name: Last activity
> tap_action:
>   action: none
> hold_action:
>   action: none

Sadly it is broken after updating to 0.6.x, any idea?

Thanks!

**Comment by zzaal** [comment]
> > image_address: https://processor.smartcamera.api.io.mi.com/miot/camera/app/v1/img?data=***
> > entity_picture: /api/camera_proxy/camera.loock_v02_1ae3_video_doorbell?token=***
> 
> You can add a picture card to display the motion picture of the video doorbell.

Viewing by reference is obtained at the beginning, once.
Then token=*** and date=*** change every few minutes.

**Comment by al-one** [comment]
Access `entity_picture` like this `http://192.168.1.100:8123/api/camera_proxy/camera.loock_v02_1ae3_video_doorbell?token=***`.
If an error is returned, check the [logs](https://my.home-assistant.io/redirect/logs).

**Comment by al-one** [comment]
> HTTPSConnectionPool(host='business.smartcamera.api.io.mi.com', port=443): Read timed out. (read timeout=10)
> Failed to establish a new connection: [Errno -3] Try again

You need to check the network status of your HA.

**Comment by al-one** [comment]
The `picture-entity` card will automatically update the latest address. If you are using in automation, you can use template:
```jinja2
{{ state_attr('camera.loock_v02_1ae3_video_doorbell', 'entity_picture') }}
```

**Comment by zzaal** [comment]
ok
how to specify the path in the card in lovelace?

**Comment by zzaal** [comment]
i test in browser:
http://192.168.1.12:8123/api/camera_proxy/camera.loock_v02_1ae3_video_doorbell?token=***
in 0.5.8. work (i see pic in browser)
in 0.6.2. NOT work (500 Internal Server Error. Server got itself in trouble)

In automation: (notification to phone)
image: "{{ state_attr('camera.loock_v02_1ae3_video_doorbell', 'entity_picture') }}"
in 0.5.8. work (i see pic in notification message)
in 0.6.2. NOT work (message without pic)

0.6.1. and 0.6.2 are broken.

**Comment by al-one** [comment]
Fixed in https://github.com/al-one/hass-xiaomi-miot/commit/b24f07f09df6a2d7860f550420d61d72c1699e4f, please upgrade to master branch and try.

**Comment by zzaal** [comment]
> Fixed in [b24f07f](https://github.com/al-one/hass-xiaomi-miot/commit/b24f07f09df6a2d7860f550420d61d72c1699e4f), please upgrade to master branch and try.

Sorry. I am newbie)
I don't know how to upgrade to master branch.
but I saw your changes in the manifest.json file and manually added the changes to my file.

Everything is working!!!
Thank you very much for your hard work and efficiency!
I'm happy!!! :)

**Comment by zzaal** [comment]
Can I ask another question?
in camera.loock_v02_1ae3_video_doorbell attributes there is another line:
stream_address: ttps://business.smartcamera.api.io.mi.com/common/app/m3u8?data=***
Entering this link in the browser downloads the file: 1651946965111.m3u8.
When viewing in vlc player on computer, i see "last live video".
How can I add it to Lovelace?
